pyperformance/data-files/benchmarks/bm_pypdf/run_benchmark.py

```diff
@@ -0,0 +1,89 @@
+"""
+A simple pypdf benchmark.
+
+Adapted from pypdf's own benchmarks:
+
+   https://github.com/py-pdf/pypdf/blob/main/tests/bench.py
+"""
+
+import io
+from pathlib import Path
+
+
+from pypdf import PdfReader, PdfWriter, Transformation
+import pyperf
+
+
+DATA_DIR = Path(__file__).parent / "data"
+
+
+def page_ops(stream, password):
+    reader = PdfReader(stream)
+    writer = PdfWriter()
+
+    if password:
+        reader.decrypt(password)
+
+    page = reader.pages[0]
+    writer.add_page(page)
+
+    op = Transformation().rotate(90).scale(1.2)
+    page.add_transformation(op)
+    page.merge_page(page)
+
+    op = Transformation().scale(1).translate(tx=1, ty=1)
+    page.add_transformation(op)
+    page.merge_page(page)
+
+    op = Transformation().rotate(90).scale(1).translate(tx=1, ty=1)
+    page.add_transformation(op)
+    page.merge_page(page)
+
+    page.add_transformation((1, 0, 0, 0, 0, 0))
+    page.scale(2, 2)
+    page.scale_by(0.5)
+    page.scale_to(100, 100)
+
+    page = writer.pages[0]
+    page.compress_content_streams()
+    page.extract_text()
+
+
+def bench_page_ops(loops):
+    """
+    Apply various page operations.
+
+    Rotation, scaling, translation, content stream compression, text extraction
+    """
+    content = (DATA_DIR / "libreoffice-writer-password.pdf").read_bytes()
```

Comment:
FWIW, GitHub says this is an invalid PDF file, though I'm sure it's fine.

Also, it may be worth adding a note about where this file came from and what's in it, especially since it's a binary format.

Reply:
Yeah, that's a good point.  It's a password-protected PDF file, which is probably why Github doesn't like it.  I just grabbed what pypdf was using -- we could choose something more deliberately.